open_optical_gating/cli/cli.py

```diff
@@ -434,6 +472,15 @@ def clear_state(self):
             (self.settings["frame_buffer_length"], self.width, self.height),
             dtype=self.dtype,
         )
+        # TODO: JT writes: I don't like this logic - I don't feel this is the right place for it.
+        # Also, update_after_n_triggers is one reason why we might want to reset the sync,
+        # but the user should have the ability to reset the sync through the GUI, or there might
+        # be other future reasons we might want to reset the sync (e.g. after each stack).
+        # I think this could partly be tidied by making self.state behave more like a proper finite state machine,
+        # A proper FSM would respond to the "reset" *input* slightly differently when in the "0" or "3" *state*.
+        # A simpler tweak that is more consistent with the current design would just involve having two different states,
+        # "reset for POG" and "reset for LTU". Both could call through to this same function, but then this function
+        # knows what it should be doing.

```

Comment:
I'm not sure I appreciate why you want to split that? In both cases we do the same thing (in my mind) - so have two different states?

Reply:
Yeah, I don't agree with myself :-) 
I'm not keen on quite how the current code does it, but my suggestion is no less clunky. Will try and come up with a better suggestion.